### PR TITLE
Added Scala 2.12 support

### PR DESCRIPTION
**Pull request by rozza**
SCALA-209

### CODE REVIEW

File: driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala [hunk 1]
```diff
@@ -19,7 +19,7 @@ package org.mongodb.scala.gridfs
 import java.io.{ByteArrayInputStream, ByteArrayOutputStream, File, InputStream}
 import javax.xml.bind.DatatypeConverter
 
-import scala.collection.JavaConversions._
```
Comment (rozza) [inline driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala:22]: JavaConversions are deprecated in 2.12

File: driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala [hunk 2]
```diff
@@ -72,10 +72,10 @@ class GridFSSpec extends RequiresMongoDBISpec {
     if (filesDocuments.nonEmpty) filesCollection.map(_.insertMany(filesDocuments)).get.futureValue
     if (chunksDocuments.nonEmpty) chunksCollection.map(_.insertMany(chunksDocuments)).get.futureValue
 
-    for (fileToArrange <- arrange.getArray("data", new BsonArray)) {
+    for (fileToArrange <- arrange.getArray("data", new BsonArray).asScala) {
```
Comment (evanchooly) [inline driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala:75]: why all the `asScala` additions?

File: .travis.yml [hunk 3]
```diff
@@ -20,6 +23,9 @@ notifications:
 
 matrix:
     fast_finish: true
+    exclude:
+      - scala: "2.12.0"
```
Comment (jyemin) [inline .travis.yml:27]: 2.12 requires Java 8 I gather?

File: driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala [hunk 4]
```diff
@@ -249,23 +249,23 @@ class GridFSSpec extends RequiresMongoDBISpec {
...
     }
     documents.toList
   }
 
   private def processChunks(bsonArray: BsonArray): List[Document] = {
     val documents = ListBuffer[Document]()
-    for (rawDocument <- bsonArray.getValues) {
-      if (rawDocument.isDocument) documents.add(parseHexDocument(rawDocument.asDocument))
+    for (rawDocument <- bsonArray.getValues.asScala) {
+      if (rawDocument.isDocument) documents += parseHexDocument(rawDocument.asDocument)
```
Comment (evanchooly) [inline driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala:268]: is this change backwards compat with 2.11 or do we care?

**Review by jyemin — APPROVED** [review]
lgtm

File: .travis.yml (hunk 3 above)
Comment (rozza) [reply .travis.yml:27]: Yeap :)

File: driver/src/test/scala/org/mongodb/scala/MongoDriverInformationSpec.scala [hunk 5]
```diff
@@ -31,7 +31,7 @@ class MongoDriverInformationSpec extends FlatSpec with Matchers {
     val exclusions = Set("$VALUES", "valueOf", "values")
 
     val wrapped = (wrappedFields ++ wrappedMethods) -- exclusions
-    val local = MongoDriverInformation.getClass.getDeclaredMethods.map(_.getName).toSet -- Set("apply")
+    val local = MongoDriverInformation.getClass.getDeclaredMethods.map(_.getName).toSet -- Set("apply", "$deserializeLambda$", "$anonfun$fromString$1")
```
Comment (evanchooly) [inline driver/src/test/scala/org/mongodb/scala/MongoDriverInformationSpec.scala:34]: This seems like a good candidate for refactoring into a common method...

**Review by evanchooly — APPROVED** [review]
otherwise, lgtm

File: driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala (hunk 2 above)
Comment (rozza) [reply driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala:75]: The automatic implicit converters are deprecated in 2.12 - this uses the explicit implicit converters!

File: driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala (hunk 4 above)
Comment (rozza) [reply driver/src/it/scala/org/mongodb/scala/gridfs/GridFSSpec.scala:268]: No - I just changing the implicit conversion

File: driver/src/test/scala/org/mongodb/scala/MongoDriverInformationSpec.scala (hunk 5 above)
Comment (rozza) [reply driver/src/test/scala/org/mongodb/scala/MongoDriverInformationSpec.scala:34]: Potentially one for the future.

**Pull request closed without merging**